sql/core/src/test/resources/sql-tests/inputs/pgSQL/strings.sql

```diff
@@ -653,8 +653,9 @@ SELECT trim(binary('\\000') from binary('\\000Tom\\000'));
 -- SELECT btrim(E'\\000trim\\000'::bytea, ''::bytea);
 -- [SPARK-28121] decode can not accept 'escape' as charset
 -- [SPARK-28412][SQL] ANSI SQL: OVERLAY function support byte array
--- SELECT encode(overlay(E'Th\\000omas'::bytea placing E'Th\\001omas'::bytea from 2),'escape');
--- SELECT encode(overlay(E'Th\\000omas'::bytea placing E'\\002\\003'::bytea from 8),'escape');
--- SELECT encode(overlay(E'Th\\000omas'::bytea placing E'\\002\\003'::bytea from 5 for 3),'escape');
```

Comment:
Why the tests are different? Those tests were ported from PostgresSQL's so we should better keep it consistent.

Reply:
Because `Spark SQL` not support `::bytea` and  encode with escape.
Should I change the tests as 
`SELECT overlay(binary('Th\\000omas') PLACING binary('Th\\001omas') FROM 2);`

Reply:
I think the expected SQL is:
```sql
SELECT encode(overlay(binary(E'Th\\000omas') placing binary(E'Th\\001omas') from 2),'escape');
```
But we don't support **String Constants with C-style Escapes**(SPARK-29034) and **`decode` can not accept 'escape' as charset**(SPARK-28121) now.

Reply:
Shall we wait for two JIRAs and enable this tests later? Once we fix the tests differently from PostgreSQL, it will be more difficult to find and fix later each instance, and we can't measure the feature party coverage properly and officially.

Reply:
@HyukjinKwon OK. please suspend this PR.

Reply:
@HyukjinKwon I read the source code again and find `String Constants with C-style Escapes` seems supported now. I have two comment in the SPARK-29034.

Reply:
It seems we still do not support `[SPARK-28121] decode can not accept 'escape' as charset`.